CB-16771 Create and include clients in CB context to communicate with…

horadla23 opened the pull request:
… Jackalope services

- added new sdx-saas-connector module for SAAS related SDX clients (client configs, proto files, etc.)
- added mock service for SDX Saas service

See detailed description in the commit message.

bergerdenes commented on sdx-connector/src/main/java/com/sequenceiq/cloudbreak/saas/client/sdx/config/SdxSaasChannelConfig.java:
```diff
@@ -0,0 +1,35 @@
+    @Value("${saas.sdx.host:localhost}")
+    private String endpoint;
+
+    @Value("${saas.sdx.port:8982}")
```
isn't there a port conflict?

horadla23 replied:
this is for the mock which is the same. there is no counterpart currently, since i am not aware of any deployment yet regarding saas. so configuration probably will happen later for this one.
but work still in prpgress

horadla23 commented:
TODO: UT for new services, classes

horadla23 commented:
TODO: UT for new services, classes

horadla23 commented:
TODO add entitlement

horadla23 commented:
- added new sdx-saas-connector module for SAAS related SDX clients (client configs, proto files, etc.)
- added mock service for SDX Saas service
- added new service to make appropriate call regarding SDX based on CRN
- refactored SDX deletion polling in environment service to use new connector
- added new field to Environment entity to differentiate platform (hybrid(saas+pass), paas)
- added new field to environment create request to indicate if the environment is hybrid or not
- added new mock test to cover hybrid use case (environment paas creation, sdx saas creation, environment cascade deletion)

lnardai commented on environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentResourceDeletionService.java:
```diff
@@ -18,20 +18,21 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(EnvironmentResourceDeletionService.class);
 
-    private final SdxEndpoint sdxEndpoint;
+    private final CrnAwareSdxConnector crnAwareSdxConnector;
```
Wouldn't this be PlatformAwareSdxConnector?

horadla23 replied:
yes, we can name it like that too. I mean different crn type means different platform so yes, they have the same meaning in our crn world.

lnardai commented on environment/src/main/resources/schema/app/20220509121640_CB-16771_add_platform_column.sql:
```diff
@@ -0,0 +1,9 @@
+-- // CB-16771 introduce platform to differentiate PaaS and hybrid setup (PaaS Env + SaaS SDX)
+-- Migration SQL that makes the change goes here.
+
+ALTER TABLE environment ADD COLUMN IF NOT EXISTS platform VARCHAR(255);
```
I don't have any better but for me platform is overloaded with cloud platform. 
We should have something unique that easy to distinguish.
deploymentType, underPinning, bedrock, foundation

lnardai commented on environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/request/EnvironmentRequest.java:
```diff
@@ -123,6 +123,9 @@ public class EnvironmentRequest extends EnvironmentBaseRequest implements Creden
     private CcmV2TlsType ccmV2TlsType;
 
+    @ApiModelProperty(EnvironmentModelDescription.HYBRID_PLATFORM)
+    private Boolean hybridPlatform;
```
Why do we need to specify the platform here? Isn't the CRN enough?

bergerdenes commented on environment/src/main/java/com/sequenceiq/environment/environment/domain/EnvironmentPlatform.java:
```diff
@@ -0,0 +1,8 @@
+public enum EnvironmentPlatform {
+    // everything is PaaS
+    PAAS,
+    // SDX SaaS, others PaaS
```
might not be granular enough in the future

horadla23 replied:
yes, I discussed it with @lnardai , I'll remove this completely, not needed

bergerdenes commented on sdx-connector/src/main/java/com/sequenceiq/cloudbreak/saas/sdx/SaasSdxService.java:
```diff
@@ -0,0 +1,84 @@
+        if (isPlatformEntitled(Crn.safeFromString(sdxCrn).getAccountId())) {
+            LOGGER.info("Calling deleteInstance for SDX SaaS instance {}", sdxCrn);
+            grpcSdxSaasClient.deleteInstance(MDCUtils.getRequestId(), sdxCrn);
+        }
```
please log if not entitled

horadla23 replied:
it is logged in EntitlementService, it would be a duplicated log

bergerdenes commented on sdx-connector/src/main/java/com/sequenceiq/cloudbreak/saas/sdx/SaasSdxService.java:
```diff
@@ -0,0 +1,84 @@
+                    .map(SDXSvcAdminProto.Instance::getCrn)
+                    .collect(Collectors.toSet());
+        }
+        return Set.of();
```
please log if not entitled

horadla23 replied:
it is logged in EntitlementService, it would be a duplicated log

bergerdenes commented on sdx-connector/src/main/java/com/sequenceiq/cloudbreak/saas/sdx/SaasSdxService.java:
```diff
@@ -0,0 +1,84 @@
+                    .map(instance -> Pair.of(instance.getCrn(), instance.getStatus()))
+                    .collect(Collectors.toSet());
+        }
+        return Set.of();
```
please log if not entitled

horadla23 replied:
it is logged in EntitlementService, it would be a duplicated log

bergerdenes commented on sdx-connector/src/main/java/com/sequenceiq/cloudbreak/saas/sdx/SaasSdxService.java:
```diff
@@ -0,0 +1,84 @@
+            return grpcSdxSaasClient.listInstances(MDCUtils.getRequestId(), environmentCrn).stream()
+                    .filter(instance -> instance.getEnvironmentsList().contains(environmentCrn));
+        }
+        return Stream.empty();
```
please log if not entitled

horadla23 replied:
it is logged in EntitlementService, it would be a duplicated log

horadla23 commented on environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentCreationDto.java:
```diff
@@ -60,6 +60,8 @@ public class EnvironmentCreationDto {
 
     private final String proxyConfigName;
 
+    private final Boolean hybridPlatform;
```
yes, I discussed it with @lnardai , I'll remove this completely, not needed

horadla23 commented:
commit message got actualized

horadla23 commented:
just rerequested review in order to make PR appear in master list:)
